test/jdk/jdk/security/jarsigner/JarWithOneNonDisabledDigestAlg.java

```diff
@@ -27,68 +27,102 @@
+        // Sanity check: Assert that MD5 is disabled, SHA-256 enabled
+        checkDigestAlgorithmPermits();
 
-        genkeypair("-alias SHA1 -sigalg SHA1withRSA");
-        genkeypair("-alias SHA256 -sigalg SHA256withRSA");
+        // Create an unsigned JAR with a single file
+        Files.write(TESTFILE1, TESTFILE1.toString().getBytes());
+        JarUtils.createJarFile(UNSIGNED_JAR, CURRENT_DIR, TESTFILE1);
 
-        KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
-        try (FileInputStream fis = new FileInputStream("keystore")) {
-            ks.load(fis, PASS.toCharArray());
-        }
+        // Generate a keystore with two different signers
+        genkeypair("-alias SIGNER1");
+        genkeypair("-alias SIGNER2");
+        KeyStore ks = loadKeyStore();
 
         // Sign JAR twice with same signer but different digest algorithms
         // so that each entry in manifest file contains two digest values.
-        signJarFile(ks, "SHA1", "MD5", "unsigned.jar", "signed.jar");
-        signJarFile(ks, "SHA1", "SHA1", "signed.jar", "signed2.jar");
-        checkThatJarIsSigned("signed2.jar", false);
+        // Note that MD5 is a disabled digest algorithm, while SHA-256 is not
+        signJarFile(ks, "SIGNER1", "MD5", UNSIGNED_JAR, SIGNED_JAR);
+        signJarFile(ks, "SIGNER1", "SHA256", SIGNED_JAR, SIGNED_TWICE_JAR);
+        checkThatJarIsSigned(SIGNED_TWICE_JAR, Map.of(TESTFILE1.toString(), 1));
 
         // add another file to the JAR
-        Files.write(Path.of(TESTFILE2), "testFile2".getBytes());
-        JarUtils.updateJarFile(Path.of("signed2.jar"), Path.of("."),
-            Path.of(TESTFILE2));
+        Files.write(TESTFILE2, TESTFILE2.toString().getBytes());
+        JarUtils.updateJarFile(SIGNED_TWICE_JAR, CURRENT_DIR, TESTFILE2);
+
+        // Sign the updated JAR, now with a different signer and with an enabled digest alg
+        signJarFile(ks, "SIGNER2", "SHA256", SIGNED_TWICE_JAR, MULTI_SIGNED_JAR);
 
-        // Sign again with different signer (SHA256) and SHA-1 digestalg.
-        // TESTFILE1 should have two signers and TESTFILE2 should have one
-        // signer.
-        signJarFile(ks, "SHA256", "SHA1", "signed2.jar", "multi-signed.jar");
+        // TESTFILE1 should have two signers and TESTFILE2 should have one signer.
+        checkThatJarIsSigned(MULTI_SIGNED_JAR,
+                Map.of(TESTFILE1.toString(), 2,
+                        TESTFILE2.toString(), 1)
+        );
+    }
 
-        checkThatJarIsSigned("multi-signed.jar", true);
+    private static void checkDigestAlgorithmPermits() throws Exception {
+        ConstraintsParameters cp = new JarConstraintsParameters(Collections.emptyList(), new Date());
+        DisabledAlgorithmConstraints jarConstraints = DisabledAlgorithmConstraints.jarConstraints();
+        try {
+            jarConstraints.permits("MD5", cp, false);
+            throw new Exception("This test assumes that MD5 is disabled");
```

Comment:
It might be clearer to say something like "This test requires MD5 to be disabled but it is enabled." 

Same for the SHA256 check below.

Reply:
Thanks!. This PR was integrated yesterday, I'll keep this in mind if I do further work in this area.